fuzzing/config/config.go

```diff
@@ -112,11 +109,47 @@ type FuzzingConfig struct {
 	TestChainConfig config.TestChainConfig `json:"chainConfig"`
 }
 
-// fuzzingConfigMarshaling is a structure that overrides field types during JSON marshaling. It allows FuzzingConfig to
-// have its custom marshaling methods auto-generated and will handle type conversions for serialization purposes.
-// For example, this enables serialization of big.Int but specifying a different field type to control serialization.
-type fuzzingConfigMarshaling struct {
-	TargetContractsBalances []*hexutil.Big
+type ContractBalance struct {
+	big.Int
+}
+
+func (cb *ContractBalance) UnmarshalJSON(data []byte) error {
+	s := strings.Trim(string(data), "\"")
+
+	if s == "null" || s == "" {
+		cb.Int.SetInt64(0)
+		return nil
+	}
+
+	// hex handling
+	if strings.HasPrefix(strings.ToLower(s), "0x") {
+		if _, ok := cb.Int.SetString(s[2:], 16); ok {
+			return nil
+		}
+		return fmt.Errorf("Error: innvalid hex string: %s", s)
+	}
+
+	// scientific notation handling
+	if strings.ContainsAny(s, "eE") {
+		f, err := strconv.ParseFloat(s, 64)
+		if err == nil {
+			plainStr := strconv.FormatFloat(f, 'f', 0, 64)
+			if _, ok := cb.Int.SetString(plainStr, 10); ok {
+				return nil
+			}
+		}
+	}
+
+	// base-10 string
+	if _, ok := cb.Int.SetString(s, 10); ok {
+		return nil
+	}
+
+	return fmt.Errorf("invalid number format: %s", s)
```

Comment:
i would highlight, as mentioned above, that this occurred while unmarshaling a contract balance. it provides the user more context.